localstack-core/localstack/services/apigateway/next_gen/execute_api/router.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+from rolo import Request, Router
+from rolo.dispatcher import Handler
+from werkzeug.routing import Rule
+
+from localstack.constants import AWS_REGION_US_EAST_1, DEFAULT_AWS_ACCOUNT_ID
+from localstack.http import Response
+from localstack.services.apigateway.models import ApiGatewayStore, apigateway_stores
+from localstack.services.edge import ROUTER
+
+from .context import RestApiInvocationContext
+from .gateway import RestApiGateway
+
+LOG = logging.getLogger(__name__)
+
+
+# TODO: subclass from -ext, and this handler will do the dispatching between v1 and v2 Gateways
+class ApiGatewayHandler:
```

Comment:
nit: As discussed, we could maybe rename this to `Endpoint` or `EndpointHandler` to differentiate between `HandlerChain` handlers here (since this is what's invoked by the router on a URL match).